Interix 3.5 does link fine with some libraries removed from the link
line (who knows why), but executing vim failes with unresolved
references to things like _dlclose, _iconv and such...

diff -ru vim71.orig/src/link.sh vim71/src/link.sh
--- vim71.orig/src/link.sh	Wed Feb 27 15:29:29 2008
+++ vim71/src/link.sh	Wed Feb 27 15:32:07 2008
@@ -41,7 +41,7 @@
   if sh link.cmd; then
     touch auto/link.sed
     cp link.cmd linkit.sh
-    for libname in SM ICE nsl dnet dnet_stub inet socket dir elf iconv Xt Xmu Xp Xpm X11 Xdmcp x w dl pthread thread readline m perl crypt attr; do
+    for libname in dummy; do
       cont=yes
       while test -n "$cont"; do
         if grep "l$libname " linkit.sh >/dev/null; then
